iib/workers/tasks/build.py

```diff
@@ -439,24 +445,20 @@ def _push_image(request_id, arch):
     :param str arch: the architecture of the container image to push
     :raises IIBError: if the push fails
     """
-    source = _get_local_pull_spec(request_id, arch)
+    source = _get_local_pull_spec(request_id, arch, include_transport=True)
     destination = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
     log.info('Pushing the container image %s to %s', source, destination)
-    run_cmd(
-        ['podman', 'push', '-q', source, destination],
+    # Use "skopeo copy" because "podman push" doesn't support setting the format
+    _skopeo_copy(
+        source,
+        destination,
         exc_msg=f'Failed to push the container image to {destination} for the arch {arch}',
     )
 
-    log.debug(f'Verifying that {destination} was pushed as a v2 manifest due to RHBZ#1810768')
+    log.debug(f'Verifying that {destination} was pushed as a v2 manifest')
     skopeo_raw = skopeo_inspect(destination, '--raw')
     if skopeo_raw['schemaVersion'] != 2:
```

Comment:
I don't think it will fallback if we specify the format. I think the check is needed to help us debug unexpected issues. If it's not schema version 2, then it'll fail later when creating the manifest list but it'll be much harder to debug it then. I'll add a comment explaining this.